### PR TITLE
Add forms SDK

### PR DESCRIPTION
**Pull request by ori-shalom**
- [x] Add `forms` SDK for handling forms and tokenize data with vault.
- [x] Optimize bundle for browsers - single file, minified output.
- [x] Push to CDN
- [x] Add README
- [x] Add a test 

Usage:
```html
<script src="https://cdn.piiano.com/public/pvault-forms-v1-0-21.js"></script>
<form action="/cards" method="post"
      data-piiano-proxy-options='{
        "collection":"credit_cards",
        "hijack":true,
        "vaultURL":"https://xxxxxxxxxx.us-east-2.awsapprunner.com",
        "apiKey": "***"
      }'>
  <h1>Add payment details</h1>
  <label for="card_number">Card Number</label>
  <input type="text" id="card_number" name="card_number"/>
  <label for="card_holder">Card Holder</label>
  <input type="text" id="card_holder" name="card_holder"/>
  <label for="card_expiry">Card Expiry</label>
  <input type="text" id="card_expiry" name="card_expiry"/>
  <label for="card_cvv">Card CVV</label>
  <input type="text" id="card_cvv" name="card_cvv"/>
  <button type="submit">Add card</button>
</form>
```

### CODE REVIEW

**Review by david-piiano — APPROVED** [review]

File: sdk/forms/vite.config.ts
Comment (nirhaas) [inline sdk/forms/vite.config.ts:1]: I have 0 exp with vite, so I hope it works :)

File: sdk/forms/test/forms.spec.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+})
+
+test.afterAll(async () => {
+  await server.close()
+})
+
+test('submitting the forms tokenize the data', async ({ page }) => {
+  await page.goto(url);
+
+  const cardNumberInput =page.getByLabel('Card Number')
```
Comment (nirhaas) [inline sdk/forms/test/forms.spec.ts:32]: Formatter?

File: sdk/forms/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+async function getAPIKey(apiKey: string | Promise<string> | (() => Promise<string>) | (() => string) | undefined): Promise<string | undefined> {
+  if (typeof apiKey === 'function') {
+    apiKey = apiKey();
+  }
+
+  return apiKey;
+}
+
+async function send({strategy = 'multi-token', object, client, tenantId, reason = 'AppFunctionality', collection}: Omit<ConfigFormOptions, 'form' | 'apiKey' | 'vaultURL' | 'onSubmit'> & { client: VaultClient, object: Record<string, any> }) {
```
Comment (nirhaas) [inline sdk/forms/src/index.ts:92]: Only `single-token` is tested, right?

File: .github/workflows/build-test-publish-package.yml [hunk 3]
```diff
@@ -81,14 +81,23 @@ jobs:
           VAULT_VERSION: ${{ inputs.vault_version }}
           PVAULT_SERVICE_LICENSE: ${{ secrets.PVAULT_SERVICE_LICENSE }}
 
-      - name: Publish package
+      - name: Configure aws credentials
+        if: inputs.bump != 'none' && github.ref_name == 'main'
+        uses: aws-actions/configure-aws-credentials@v2
```
Comment (nirhaas) [inline .github/workflows/build-test-publish-package.yml:86]:
```suggestion
        uses: aws-actions/configure-aws-credentials@v4
```

File: .github/workflows/build-test-publish-package.yml [hunk 4]
```diff
@@ -81,14 +81,23 @@ jobs:
...
+        with:
+          role-to-assume: arn:aws:iam::032109477184:role/vault-saas-prod-piiano-vault-typescript-github-actions
```
Comment (nirhaas) [inline .github/workflows/build-test-publish-package.yml:88]: Why are you assuming this role?

File: sdk/forms/test/forms.spec.ts (hunk 1 above)
Comment (ori-shalom) [reply sdk/forms/test/forms.spec.ts:32]: None of the projects in this repo has prettier or even eslint at the moment.
I'll manually format it and we should add it later for all projects.

File: sdk/forms/src/index.ts (hunk 2 above)
Comment (ori-shalom) [reply sdk/forms/src/index.ts:92]: Only `multi-token` is tested and actually I think currently the others won't work.
This is just a first version and focus is on creating the baseline that we can enrich later.

File: .github/workflows/build-test-publish-package.yml (hunk 4 above)
Comment (ori-shalom) [reply .github/workflows/build-test-publish-package.yml:88]: To push the forms code to s3 of the CDN.
You can look at it `package.json` `push` command.

File: sdk/forms/src/styles.css [hunk 5]
```diff
@@ -0,0 +1,23 @@
+body {
```
Comment (nirhaas) [inline sdk/forms/src/styles.css:1]: Why is this file here

Comment (ori-shalom) [reply sdk/forms/src/styles.css:1]: It's not packaged, it's just used for the local dev page to look nice.
Original thought was we might enrich this page to be a full demo but probably a demo app will be another project...
